src/ota_stm32f.c

```diff
@@ -0,0 +1,223 @@
+#include "flash.h"
+#include "log.h"
+#include "ota.h"
+
+#if MG_OTA == MG_OTA_STM32F
+
+static bool mg_stm32f_write(void *, const void *, size_t);
+static bool mg_stm32f_swap(void);
+
+static struct mg_flash s_mg_flash_stm32f = {
+    (void *) 0x08000000,  // Start
+    0,                    // Size, FLASH_SIZE_REG
+    0,                    // Irregular sector size
+    32,                   // Align, 256 bit
+    mg_stm32f_write,
+    mg_stm32f_swap,
+};
+
+#define FLASH_BASE_REGS 0x40023c00  // Base address for flash interface
```

Comment:
Please keep a consistent naming across variants, it helps maintenance. For all three H5 H7 and Fx this is the start of the flash controller block of registers, yet we name then differently.
_BASE is OK, just make it clear on the comment; in the rest of the code, when one sees `MG_REG(FLASH_BASE + FLASH_KEYR)` it is obvious that we're not talking about the start of flash memory

Reply:
Just changed it now to something different than FLASH_BASE because some cmsis header files already define FLASH_BASE (I saw that on F207 Freertos).

Reply:
Good point
I have a note of something similar and I didn't take action yet.
In these cases I think it is better to prepend MG_...

Reply:
Though I'd rather have a consistent naming scheme, it helps when coming back after a while or comparing to other parts

Reply:
We'll use MG_FLASH* then (MG_FLASH_BASE, MG_FLASH_KEYR, etc...)